include/field2d.hxx

```diff
@@ -111,6 +111,24 @@ class Field2D : public Field, public FieldData {
    */
   int getNz() const override {return 1;};
 
+  /*!
+   * Return the coordinate system of this field
+   */
+  COORDINATE_SYSTEM getCoordinateSystem() const {
+#if CHECK > 0
```

Comment:
I think this should always throw as it's a sign of an invalid executable and there's no performance penalty (i.e. it's not a check that typically passes in a valid program).